controller/src/main/java/io/pravega/controller/server/rest/RESTServer.java

```diff
@@ -68,12 +74,39 @@ protected void startUp() {
         long traceId = LoggerHelpers.traceEnterWithContext(log, this.objectId, "startUp");
         try {
             log.info("Starting REST server listening on port: {}", this.restServerConfig.getPort());
-            httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            if (restServerConfig.isTlsEnabled()) {
+                SSLContextConfigurator contextConfigurator = new SSLContextConfigurator();
+                contextConfigurator.setKeyStoreFile(restServerConfig.getKeyFilePath());
+                contextConfigurator.setKeyStorePass(loadPasswordFromFile(restServerConfig.getKeyFilePasswordPath()));
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true,
+                        new SSLEngineConfigurator(contextConfigurator, false, false, false));
+            } else {
+                httpServer = GrizzlyHttpServerFactory.createHttpServer(baseUri, resourceConfig, true);
+            }
         } finally {
             LoggerHelpers.traceLeave(log, this.objectId, "startUp", traceId);
         }
     }
 
+    private String loadPasswordFromFile(String keyFilePasswordPath) {
+        // In case the path is not specified, return empty string. This means the password is not used.
+        // In case of error, return empty password. Which will fail the SSL connection if the password is expected.
+
+        if (Strings.isNullOrEmpty(keyFilePasswordPath)) {
+            return "";
+        }
+        File passwdFile = new File(keyFilePasswordPath);
+        if (passwdFile.length() == 0) {
+            return "";
+        }
+        try {
+            return new String(FileUtils.readFileToByteArray(passwdFile)).trim();
```

Comment:
You need to specify the encoding, otherwise this will try to decode the file using the default charset, which may not match what's in the file.